### PR TITLE
ESP interaction

### PR DESCRIPTION
**Pull request by lochieferrier**
A short function for ESP .csm file re-writing.

I have never done a contribution to GPkit so I apologies in advance for screwing up some part of the process.

### CODE REVIEW

**Comment by bqpd** [comment]
add to whitelist

File: gpkit/interactive/rendering.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+"""Rendering methods"""
+
+"""Save a set of variables to a .csm, for usage
```
Comment (bqpd) [inline gpkit/interactive/rendering.py]: Docstring should be on the line below function declaration, with the following formatting:

```
"""Save a set of variables to a .csm, for usage with the Engineering Sketch Pad rendering software.
 
Arguments
---------
filepath : string
    the full filepath to the .csm, for example "design.csm" or "renders/version1.csm"

sol : SolutionArray
 
paramdict : dict
    {design parameter string in ESP: corresponding Variable}
 """
```

File: gpkit/interactive/rendering.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+- sol is a GPkit Solution instance
+
+- paramdict is a dictionary comprised of GPkit Variables
+indexed by their design parameter name in ESP (String).
+"""
+
+def saveVariablesToCSM(filepath,sol,paramdict):
+        contents = ""
+        if filepath != "":
```
Comment (bqpd) [inline gpkit/interactive/rendering.py]: if filepath is an optional argument, function signature should be `sol, paramdict, filepath=None`, and this conditional should be `if filepath is not None`

File: gpkit/interactive/rendering.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+- filepath is a string with the full filepath to the .csm,
+for example "design.csm" or "renders/version1.csm"
+
+- sol is a GPkit Solution instance
+
+- paramdict is a dictionary comprised of GPkit Variables
+indexed by their design parameter name in ESP (String).
+"""
+
+def saveVariablesToCSM(filepath,sol,paramdict):
```
Comment (bqpd) [inline gpkit/interactive/rendering.py]: lint: spaces between arguments in function signature

File: gpkit/interactive/rendering.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        contents = ""
+        if filepath != "":
+            with open(filepath, 'r') as file:
+                contents=file.read()
+        lines = contents.splitlines()
+        for i,line in enumerate(lines):
```
Comment (bqpd) [inline gpkit/interactive/rendering.py]: `i, line`

File: gpkit/interactive/rendering.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+- paramdict is a dictionary comprised of GPkit Variables
+indexed by their design parameter name in ESP (String).
+"""
+
+def saveVariablesToCSM(filepath,sol,paramdict):
+        contents = ""
+        if filepath != "":
+            with open(filepath, 'r') as file:
+                contents=file.read()
```
Comment (bqpd) [inline gpkit/interactive/rendering.py]: spaces around the equals sign

File: gpkit/interactive/rendering.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        lines = contents.splitlines()
```
Comment (bqpd) [inline gpkit/interactive/rendering.py]: ...wait, so this'll be an empty list if `content == ""`?

File: gpkit/interactive/rendering.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+                if key in paramdict:
+                    valStr = str(float(sol(paramdict[key])))
+                    line = line[:len(line)-len(valStr)] + valStr 
+                    lines[i] = line
+        output = ""
+        for line in lines:
+            output = output + line + '\n'
+        contents = output
+        with open(filepath, "w") as file:
+            file.write(contents)
```
Comment (bqpd) [inline gpkit/interactive/rendering.py]: hmm, I'd replace these three lines with

```python
with open(filepath, "w") as file:
    file.write(output)
```

File: gpkit/interactive/rendering.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        for i,line in enumerate(lines):
+            if line[0:7] == 'despmtr':
+                key = line[10:25].strip()
+                if key in paramdict:
+                    valStr = str(float(sol(paramdict[key])))
+                    line = line[:len(line)-len(valStr)] + valStr 
+                    lines[i] = line
+        output = ""
+        for line in lines:
+            output = output + line + '\n'
```
Comment (bqpd) [inline gpkit/interactive/rendering.py]: or `output += line + "\n"`, if you feel like bein fancy

File: gpkit/interactive/rendering.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        if filepath != "":
+            with open(filepath, 'r') as file:
+                contents=file.read()
+        lines = contents.splitlines()
+        for i,line in enumerate(lines):
+            if line[0:7] == 'despmtr':
+                key = line[10:25].strip()
+                if key in paramdict:
+                    valStr = str(float(sol(paramdict[key])))
+                    line = line[:len(line)-len(valStr)] + valStr
```
Comment (bqpd) [inline gpkit/interactive/rendering.py]: equivalently:

```python
lines[i] = line[:len(line)-len(valStr)] + "%f" % (sol(paramdict[key])
```

**Comment by galbramc** [comment]
@bqpd, you were not an admin before in jeknins for gpkit pull requests. I just added you. Can you try whitelisting Lochi again to check if it works now?

**Comment by bqpd** [comment]
add to whitelist

**Pull request closed without merging**